internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -138,6 +141,11 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 func (r *Reconciler) SetupForDryRun(recorder record.EventRecorder) {
 	r.desiredStateGenerator = desiredstate.NewGenerator(r.Client, r.ClusterCache, r.RuntimeClient)
 	r.recorder = recorder
+	r.externalTracker = external.ObjectTracker{
+		Controller: fakeController.Controller{},
+		Cache:      &informertest.FakeInformers{},
+		Scheme:     runtime.NewScheme(),
```

Comment:
```suggestion
		Scheme:     r.Client.Scheme(),
```
Just to make this slightly safer (the schema from client should contain all the types we care about)